packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts

```diff
@@ -173,6 +181,23 @@ export class SageMakerCreateTransformJob extends sfn.TaskStateBase {
     };
   }
 
+  private renderModelClientOptions(config: ModelClientOptions): { [key: string]: any } {
+    const retries = config.invocationsMaxRetries;
+    if (retries? (0 >= retries || retries >= 3): false) {
+      throw new RangeError(`invocationsMaxRetries should be between 0 and 3, not ${retries}.`);
+    }
+    const timeout = config.invocationsTimeout?.toSeconds();
+    if (timeout? (1 >= timeout || timeout >= 3600): false) {
+      throw new RangeError(`invocationsTimeout should be between 1 and 3600 seconds, not ${timeout}.`);
+    }
```

Comment:
Added validation for both the properties based on what is documented [on the API page](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ModelClientConfig.html).